manim/renderer/cairo_renderer.py

```diff
@@ -74,7 +75,7 @@ def __init__(self, camera_class=None, skip_animations=False, **kwargs):
         self.original_skipping_status = skip_animations
         self.skip_animations = skip_animations
         self.animations_hashes = []
-        self.num_plays = 0
+        self.num_plays = -1
```

Comment:
Why did you do this and then change the comparison in update_skipping_status()?

Reply:
Because the first time that `handle_play_like_call` is called (i.e. the first time an animation is played), step 1 of my explanation above will try to close any open pipes _and increment the counter_.

So, I could implement a bunch of logic in multiple places to avoid incrementing the counter that first time _only_. OR I can just start the count at -1 :sweat_smile: Accordingly, `update_skipping_status` also needs to change because it's called before the increment happens to it now off-by-one.

For the record, I think this is hacky af and I hate it. But also I'm getting tired of this PR :eyes: and I think this can maybe be merged now since it doesn't break any other tests and we can deal with the nasty details later?

Reply:
Why increment the counter after the pipe is closed? Couldn't you just do it at the end of play()?

Reply:
On paper, that sounds like a good idea. I just tried it, but it's just bugging out :shrug: 

If anyone can figure out a way of implementing @eulertour 's suggestion, please go ahead.

Otherwise, I'm going to add some tests to this and call it.